test/functional/feature_elements_taproot_activation.py

```diff
@@ -0,0 +1,121 @@
+        self.log.info("Testing activation at height %d" % activation_height)
+        activation_height = 128 * ((activation_height + 127) // 128)
+
+        assert_equal(self.nodes[node_idx].getblockcount(), 0)
+
+        blocks = self.nodes[node_idx].generatetoaddress(activation_height - 2, self.nodes[node_idx].getnewaddress())
+        assert_equal(self.nodes[node_idx].getblockcount(), activation_height - 2)
+
+        for n, block in enumerate(blocks):
+            decode = self.nodes[node_idx].getblockheader(block)
+            if n < 143:
+                assert_equal (decode["versionHex"], "20000000")
+            elif n < 431:
+                # TESTDUMMY deployment: 144 blocks active, 144 blocks locked in
+                assert_equal (decode["versionHex"], "30000000")
+            else:
+                assert_equal (decode["versionHex"], "20000000")
+
+        assert_equal(self.nodes[node_idx].getblockchaininfo()["softforks"]["taproot"]["bip9"]["status"], "defined")
+        # The 1023rd block does not signal, but changes the signalling state
+        # to "started" from "defined"
+        blocks = self.nodes[node_idx].generatetoaddress(1, self.nodes[node_idx].getnewaddress())
+        assert_equal(self.nodes[node_idx].getblockchaininfo()["softforks"]["taproot"]["bip9"]["status"], "started")
+        assert_equal(self.nodes[node_idx].getblockheader(blocks[node_idx])["versionHex"], "20000000")
+
+        blocks = self.nodes[node_idx].generatetoaddress(127, self.nodes[node_idx].getnewaddress())
+        for n, block in enumerate(blocks):
+            decode = self.nodes[node_idx].getblockheader(block)
+            assert_equal (decode["versionHex"], "20000004")
+        assert_equal(self.nodes[node_idx].getblockchaininfo()["softforks"]["taproot"]["bip9"]["status"], "started")
+
+        # Fail to signal on the 128th block. Since the threshold for Taproot is
+        # 100% this will prevent activation. Note that our period is 128, not
+        # 144 (the default), as we have overridden the period for Taproot. On
+        # the main Liquid chain it is overridden to be one week of signalling.
+        block = self.nodes[node_idx].getnewblockhex()
+        block = block[:1] + "0" + block[2:] # turn off Taproot signal
+        self.nodes[node_idx].submitblock(block)
+        assert_equal(self.nodes[node_idx].getblockchaininfo()["softforks"]["taproot"]["bip9"]["status"], "started")
+
+        # Run through another 128 blocks, without failing to signal
+        blocks = self.nodes[node_idx].generatetoaddress(127, self.nodes[node_idx].getnewaddress())
+        for n, block in enumerate(blocks):
+            decode = self.nodes[node_idx].getblockheader(block)
+            assert_equal (decode["versionHex"], "20000004")
+        assert_equal(self.nodes[node_idx].getblockchaininfo()["softforks"]["taproot"]["bip9"]["status"], "started")
+        # The 128th block then switches from "locked_in" to "active"
```

Comment:
In 0df8895b0ba53871fde3d42a766442b58439e389 "test: add test for Taproot activation"

```suggestion
        # The 128th block then switches from "started" to "locked_in"
```